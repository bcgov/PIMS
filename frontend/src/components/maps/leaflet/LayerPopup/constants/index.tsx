import * as React from 'react';

export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PARCEL_NAME: { label: 'Parcel Name:', display: (data: any) => data.PARCEL_NAME },
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  PLAN_NUMBER: { label: 'Plan Number:', display: (data: any) => data.PLAN_NUMBER },
  PARCEL_CLASS: { label: 'Parcel Class:', display: (data: any) => data.PARCEL_CLASS },
  MUNICIPALITY: { label: 'Municipality:', display: (data: any) => data.MUNICIPALITY },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area:',
    display: (data: any) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
