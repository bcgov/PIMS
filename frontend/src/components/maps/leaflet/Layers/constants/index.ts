export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_NAME: { label: 'Area Name:', display: (data: any) => data.ADMIN_AREA_NAME },
  ADMIN_AREA_ABBREVIATION: {
    label: 'Area Abbrevation:',
    display: (data: any) => data.ADMIN_AREA_ABBREVIATION,
  },
  ADMIN_AREA_GROUP_NAME: {
    label: 'Area Group Name:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
