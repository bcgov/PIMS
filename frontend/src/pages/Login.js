import { Redirect } from 'react-router-dom';
import { useKeycloak } from 'react-keycloak';
import React from 'react';
import './Login.scss';

function Login() {
  const { keycloak } = useKeycloak();

  if (keycloak.authenticated) {
    return <Redirect to={{ pathname: '/mapview' }} />;
  }

  return (
    <div className="unauth">
      <h1>Welcome! Connect to Start</h1>
      <div>
        <span>
          <a className="button" onClick={() => keycloak.login()}>
            Sign-in
            </a>
        </span>
      </div>
    </div>
  );
}

export default Login;
