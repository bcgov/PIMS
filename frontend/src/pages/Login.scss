@import '../styles.scss';
@import '../colors.scss';

.unauth {
  font-size: 24px;
  margin: 91px auto;

  .row {
    justify-content: center;
  }

  .block {
    background-color: #f2f2f2;
    padding: 43px 0;
    max-width: 776px;
  }

  h1 {
    padding-bottom: 18px;
    font-family: 'BCSans', Fallback, sans-serif;
    font-weight: 700;
    font-size: 24px;
    color: $text-color;
  }

  p {
    font-size: 16px;
    color: $text-color;
    font-family: 'BCSans', Fallback, sans-serif;
    line-height: 130%;
    width: 75%;
  }

  .blockText {
    margin: 0 auto 38px auto;
  }

  .or {
    margin: 20px auto;
  }

  .border-dark {
    border-color: $primary-color !important;
    color: $text-color !important;
  }
}
