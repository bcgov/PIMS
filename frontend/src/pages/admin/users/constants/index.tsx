import * as React from 'react';
import { MUIDataTableColumnDef } from 'mui-datatables';
import { FaCheck, FaTimes } from 'react-icons/fa';
import { AccountActive } from '../interfaces/IUserRecord';
import { RowActions } from '../components/RowActions';
export const columns: MUIDataTableColumnDef[] = [
  {
    name: 'id',
    label: 'ID',
    options: {
      display: 'excluded',
      filter: false,
    },
  },
  {
    name: 'username',
    label: 'IDIR/BCeID',
    options: {
      customBodyRender: (username: string, tableMeta: any) => {
        return <a href={`/admin/user/${tableMeta.rowData[0]}`}>{username}</a>;
      },
    },
  },
  {
    name: 'firstName',
    label: 'First name',
  },
  {
    name: 'lastName',
    label: 'Last name',
  },
  {
    name: 'email',
    label: 'Email',
  },
  {
    name: 'active',
    label: 'Active',
    options: {
      customBodyRender: (active: AccountActive, tableMeta: any) => {
        return active === AccountActive.YES ? <FaCheck /> : <FaTimes />;
      },
    },
  },
  {
    name: 'agency',
    label: 'Agency',
  },
  {
    name: 'position',
    label: 'Position',
  },
  {
    name: 'role',
    label: 'Role',
  },
  {
    name: 'action',
    label: 'Actions',
    options: {
      customBodyRender: (value: any, tableMeta: any) => {
        return <RowActions userId={tableMeta.rowData[0]} active={tableMeta.rowData[5]} />;
      },
    },
  },
];
