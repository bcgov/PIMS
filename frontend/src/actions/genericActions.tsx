import * as ActionTypes from 'constants/actionTypes';

export interface IGenericNetworkAction {
  isFetching: boolean;
  name: string;
  type: string;
  errorMessage?: string;
  status?: number;
  data?: any;
}

export const success = (reducer: string, status?: number, data?: any): IGenericNetworkAction => ({
  isFetching: false,
  name: reducer,
  type: ActionTypes.SUCCESS,
  status,
  data,
});

export const request = (reducer: string): IGenericNetworkAction => ({
  isFetching: true,
  name: reducer,
  type: ActionTypes.REQUEST,
});

export const error = (reducer: string, status?: number, err?: any): IGenericNetworkAction => ({
  isFetching: false,
  name: reducer,
  type: ActionTypes.ERROR,
  errorMessage: err,
  status,
});

export const clear = (reducer: string): IGenericNetworkAction => ({
  isFetching: false,
  name: reducer,
  type: ActionTypes.CLEAR,
});
