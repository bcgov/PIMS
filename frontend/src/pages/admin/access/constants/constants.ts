import { ColDef } from 'ag-grid-community';
import { IAccessRequest } from 'interfaces';
import { AccessStatusDisplay, AccessStatusDisplayMapper } from 'constants/accessStatus';

export const columnDefinitions = (): ColDef[] => {
  let items: ColDef[] = [
    {
      headerName: 'IDIR/BCeID',
      checkboxSelection: true,
      headerCheckboxSelection: true,
      valueGetter: params => (params.data as IAccessRequest).user.username,
    },
    {
      headerName: 'First name',
      valueGetter: params => (params.data as IAccessRequest).user.firstName,
    },
    {
      headerName: 'Last name',
      valueGetter: params => (params.data as IAccessRequest).user.lastName,
    },
    {
      headerName: 'Email',
      valueGetter: params => (params.data as IAccessRequest).user.email,
      width: 250,
    },
    {
      headerName: 'Position',
      valueGetter: params => (params.data as IAccessRequest).position,
    },
    {
      headerName: 'Status',
      valueGetter: (params): AccessStatusDisplay =>
        AccessStatusDisplayMapper[(params.data as IAccessRequest).status],
      filter: false,
      sortable: false,
    },
    {
      headerName: 'Agency',
      valueGetter: params => (params.data as IAccessRequest).agencies[0].name,
      width: 250,
    },
    {
      headerName: 'Role',
      valueGetter: params => (params.data as IAccessRequest).roles[0].name,
    },
  ];

  items = items.map(column => ({
    filter: true,
    suppressMenu: true,
    cellClass: 'user-info-cell',
    sortable: true,
    minWidth: 230,
    ...column,
  }));

  return items;
};
