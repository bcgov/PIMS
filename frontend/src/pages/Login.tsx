import { Redirect } from 'react-router-dom';
import { useKeycloak } from 'react-keycloak';
import { Button } from 'react-bootstrap';
import React from 'react';
import './Login.scss';

function Login() {
  const { keycloak } = useKeycloak();

  if (keycloak.authenticated) {
    return <Redirect to={{ pathname: '/mapview' }} />;
  }

  return (
    <div className="unauth">
      <h1>Search and manage government properties</h1>
      <div className="block">
        <p className="blockText">Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          Praesent libero nibh, venenatis a ligula a, mattis porta lacus.
          Praesent sagittis sed velit eget interdum.
          In cursus pellentesque dui vitae efficitur.
        </p>
        <Button variant="primary" onClick={() => keycloak.login()}>
          Sign In
        </Button>
        <p className="or">Or</p>
        <Button className="border border-dark" variant="secondary" >
          Sign Up
        </Button>
      </div>
    </div>
  );
}

export default Login;
