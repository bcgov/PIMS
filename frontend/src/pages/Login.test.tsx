import React from 'react';
import { createMemoryHistory } from 'history';
import { render, fireEvent } from '@testing-library/react';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { useKeycloak } from '@react-keycloak/web';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import Login from '../pages/Login';
import * as reducerTypes from 'constants/reducerTypes';
import {} from 'reducers/networkReducer';
import { IGenericNetworkAction } from 'actions/genericActions';
import { Provider } from 'react-redux';

jest.mock('@react-keycloak/web');
const mockStore = configureMockStore([thunk]);

const store = mockStore({});

//boilerplate function used by most tests to wrap the Login component with a router.
const renderLogin = () => {
  const history = createMemoryHistory();
  return render(
    <Provider store={store}>
      <Router history={history}>
        <Login />
      </Router>
    </Provider>,
  );
};

test('login renders correctly', () => {
  (useKeycloak as jest.Mock).mockReturnValue({ keycloak: { authenticated: false } });
  const history = createMemoryHistory();
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router history={history}>
          <Login></Login>
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('authenticated users are redirected to the mapview', () => {
  (useKeycloak as jest.Mock).mockReturnValue({
    keycloak: { authenticated: true, realmAccess: { roles: ['property-view'] } },
  });
  const history = createMemoryHistory();

  render(
    <Provider store={store}>
      <Router history={history}>
        <Login />
      </Router>
    </Provider>,
  );
  expect(history.location.pathname).toBe('/mapview');
});

test('new users are sent to the guest page', () => {
  (useKeycloak as jest.Mock).mockReturnValue({
    keycloak: { authenticated: true, realmAccess: { roles: [{}] } },
  });
  const history = createMemoryHistory();
  const activatedAction: IGenericNetworkAction = {
    isFetching: false,
    name: 'test',
    type: 'POST',
    status: 201,
  };
  const store = mockStore({ activateUser: activatedAction });

  render(
    <Provider store={store}>
      <Router history={history}>
        <Login />
      </Router>
    </Provider>,
  );
  expect(history.location.pathname).toBe('/guest');
});

test('unAuthenticated users are shown the login screen', () => {
  (useKeycloak as jest.Mock).mockReturnValue({ keycloak: { authenticated: false } });
  const { getByRole } = renderLogin();
  expect(getByRole('heading')).toHaveTextContent('Search and manage government properties');
});

test('a spinner is displayed if keycloak has not yet been initialized', () => {
  (useKeycloak as jest.Mock).mockReturnValue({ keycloak: undefined });
  const { container } = renderLogin();
  expect(container.firstChild).toHaveClass('spinner-border');
});

test('the login button calls keycloaks login() method', () => {
  const login = jest.fn();
  (useKeycloak as jest.Mock).mockReturnValue({ keycloak: { login: login, authenticated: false } });

  const { getByText } = renderLogin();
  fireEvent.click(getByText(/sign in/i));

  expect(login.mock.calls.length).toBe(1);
});
