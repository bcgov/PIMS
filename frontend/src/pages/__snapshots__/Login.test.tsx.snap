// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login login renders correctly 1`] = `
<div
  className="unauth container-fluid"
>
  <h1>
    Search and manage government properties
  </h1>
  <div
    className="row"
  >
    <div
      className="col-md-3 col-1"
    />
    <div
      className="block col-md-6 col-16"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Sign In
      </button>
      <p
        className="or"
      >
        Or
      </p>
      <button
        className="border border-dark btn btn-secondary"
        disabled={false}
        type="button"
      >
        Sign Up
      </button>
    </div>
    <div
      className="col-md-3 col-1"
    />
  </div>
</div>
`;
